BraintreeCore/src/main/java/com/braintreepayments/api/ApiClient.kt

```diff
@@ -35,21 +29,46 @@ class ApiClient constructor(private val braintreeClient: BraintreeClient) {
         try {
             val body = paymentMethod.buildJSON().toString()
             braintreeClient.sendPOST(url, body) { responseBody, httpError ->
-                if (responseBody != null) {
-                    try {
-                        callback.onResult(JSONObject(responseBody), null)
-                    } catch (exception: JSONException) {
-                        callback.onResult(null, exception)
-                    }
-                } else {
-                    callback.onResult(null, httpError)
-                }
+                parseResponse(responseBody, httpError, callback)
             }
         } catch (exception: JSONException) {
-            callback.onResult(null, exception)
+            returnError(callback, exception)
+        }
+    }
+
+    private fun parseResponse(
+        responseBody: String?,
+        exception: Exception?,
+        callback: TokenizeCallback,
+        analyticsEventName: String? = null) {
+        responseBody?.also {
+            try {
+                returnSuccess(callback, JSONObject(it))
+                analyticsEventName?.also { event ->
+                    braintreeClient.sendAnalyticsEvent("$event.success")
+                }
+            } catch (exception: JSONException) {
+                analyticsEventName?.also { event ->
+                    braintreeClient.sendAnalyticsEvent("$event.failure")
+                }
+                returnError(callback, exception)
+            }
+        } ?: run {
+            analyticsEventName?.also { event ->
+                braintreeClient.sendAnalyticsEvent("$event.failure")
+            }
+            returnError(callback, exception)
         }
     }
 
+    private fun returnSuccess(callback: TokenizeCallback, jsonObject: JSONObject?) {
+        callback.onResult(jsonObject, null)
+    }
```

Comment:
Could we inline `callback.onResult(jsonObject, null` instead? I see the appeal for readability but the tradeoff of having `returnSuccess` and `returnError` feels more procedural than OOP since we still have to pass in the callback param.